processing-service-core/api/swagger/swagger.yaml

```diff
@@ -952,7 +954,246 @@ paths:
           description: The specified Workflow could not be found.
           schema:
             $ref: "#/definitions/errorMessage"    
+  ##########################
+  ##  Tenant Key Store
+  ##########################
+  /tenants/{tenantId}/config/{key}:
+    x-swagger-router-controller: tenant
+    parameters:
+      - name: tenantId
+        in: path
+        type: string
+        description: The ID of the tenant.
+        required: true  
+      - name: key
+        in: path
+        type: string
+        description: The key to use when searching for a tenants configuration.
+        required: true
+    put:
+      tags:
+        - Tenant
+      parameters:
+        - name: value
+          type: string
+          in: query
+          description: The value to use when setting the custom config for the tenants key.
+      summary: Creates or updates tenant specific config.
+      description: Creates or updates a custom configuration setting for this key against this tenant if one doesn't exist or updates the currently existing custom configuration setting if it does.
+      operationId: setTenantConfig
+      responses:
+        201:
+          description: Indicates that the tenant specific configuration was created successfully.
+        204:
+          description:  Indicates that the tenant specific configuration was updated successfully.
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+    get:
+      tags:
+        - Tenant
+      summary: Retrieve tenant specific config.
+      description: Returns the tenant specific config for this key that is stored against this tenant.
+      operationId: getTenantConfig
+      responses:
+        200:
+          description: Returns the tenant specific configuration stored for the supplied key.
+          schema:
+            $ref: "#/definitions/tenantConfig"
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+    delete:
+      tags:
+        - Tenant
+      summary: Delete tenant specific config for the provided key.
+      description: Delete tenant specific config for the provided key.
+      operationId: deleteTenantConfig
+      responses:
+        204:
+          description: Deletes the tenant specific config for the supplied key.
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+  #
+  /tenants/{tenantId}/:
```

Comment:
Given what these methods do this should probably be `/tenants/{tenantId}/config` rather than just `/tenants/{tenantId}`, as they only manipulate the tenant's configuration and not other things about the tenant.